test/unit/services/finance/stripe_charge_service_test.rb

```diff
@@ -105,6 +106,26 @@ class Finance::StripeChargeServiceTest < ActiveSupport::TestCase
     assert_equal 'requires_payment_method', payment_intent.reload.state
   end
 
+  test 'has payment intent description when invoice is present' do
+    invoice.update({friendly_id: '2022-10-00000001'})
+    invoice.issue_and_pay_if_free!
+    with_invoice = build_charge_service(invoice: invoice)
+    assert_equal "#{invoice.provider.name} API services 2022-10-00000001", with_invoice.send(:charge_description)
+  end
+
+  test 'has payment intent description with no invoice in charging service' do
+    without_invoice = build_charge_service
+    assert_equal 'API services', without_invoice.send(:charge_description)
+  end
+
+  test 'sends payment intent description to the gateway' do
+    response = build_response(true, 'Transaction Approved', object: 'payment_intent', id: 'new-payment-intent-id', status: 'succeeded')
+    invoice.issue_and_pay_if_free!
+    gateway.expects(:purchase).with(15000, anything, has_entry(:description, "#{invoice.provider.name} API services fix")).returns(response)
```

Comment:
I think we don't care about cops in tests but anyways, I suggest to use the big numbers convention:

```suggestion
    gateway.expects(:purchase).with(15_000, anything, has_entry(:description, "#{invoice.provider.name} API services fix")).returns(response)
```